### PR TITLE
feat: user common model endpoints, service, type, and mapper

### PR DESCRIPTION
**Pull request by asdfryan**
TODO:
- open API spec + typing
- hubspot/salesforce implementation
- owner associations

### CODE REVIEW

File: apps/api/routes/crm/v1/user.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { camelcaseKeys } from '@/lib/camelcase';
+import { snakecaseKeys } from '@/lib/snakecase';
+import { Request, Response, Router } from 'express';
+
+const { userService } = getDependencyContainer();
+
+export default function init(app: Router): void {
+  const router = Router();
+
+  router.get('/', async (req: Request, res: Response) => {
```
Comment (albertyfwu) [inline apps/api/routes/crm/v1/user.ts:11]: can we use the actual types to have type-checking?

File: packages/core/types/crm/lead.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import type { CrmAccount, CrmContact, CrmLead } from '@supaglue/db';
-import type { Account, Address, Contact, EmailAddress, PhoneNumber } from '.';
+import type { Account, Address, Contact, EmailAddress, PhoneNumber } from '../index';
```
Comment (albertyfwu) [inline packages/core/types/crm/lead.ts:2]: just `..` is enough

**Review by albertyfwu — APPROVED** [review]

File: apps/api/routes/crm/v1/user.ts (hunk 1 above)
Comment (asdfryan) [reply apps/api/routes/crm/v1/user.ts:11]: Yes I will, but I haven't done the open API spec yet so we don't have the types (next PR).

File: packages/core/types/crm/lead.ts (hunk 2 above)
Comment (asdfryan) [reply packages/core/types/crm/lead.ts:2]: will fix in next PR

**Pull request merged**